src/Authenticator/LoginForm.php

```diff
@@ -230,14 +202,75 @@ public function forTemplate()
      */
     public function RealMeLastError()
     {
-        $message = Session::get('RealMe.LastErrorMessage');
-        Session::clear('RealMe.LastErrorMessage');
+        $session = $this->getRequest()->getSession();
+
+        $message = $session->get('RealMe.LastErrorMessage');
+        $session->clear('RealMe.LastErrorMessage');
 
         return $message;
     }
 
     public function HasRealMeLastError()
     {
-        return Session::get('RealMe.LastErrorMessage') !== null;
+        return $this->getRequest()->getSession()->get('RealMe.LastErrorMessage') !== null;
+    }
+
+    /**
+     * Return the title of the form for use in the frontend
+     * For tabs with multiple login methods, for example.
+     * This replaces the old `get_name` method
+     * @return string
+     */
+    public function getAuthenticatorName()
+    {
+        return 'RealMe Login';
+    }
+
+    /**
+     * Required FieldList creation on a LoginForm
+     *
+     * @return FieldList
+     */
+    protected function getFormFields()
+    {
+        return new FieldList(array(
+            new HiddenField('AuthenticationMethod', null, $this->authenticator_class)
+        ));
+    }
+
+    /**
+     * Required FieldList creation for the login actions on this LoginForm
+     *
+     * @return FieldList
+     */
+    protected function getFormActions()
+    {
+        /** @var RealMeService $service */
+        $service = Injector::inst()->get(RealMeService::class);
+
+        $integrationType = $service->config()->integration_type;
+
+        if ($integrationType === RealMeService::TYPE_ASSERT) {
+            $loginButtonContent = ArrayData::create(array(
+                'Label' => _t(
+                    'RealMeLoginForm.ASSERTLOGINBUTTON',
+                    'Share your details with {orgname}',
+                    '',
+                    ['orgname' => $service->config()->metadata_organisation_display_name]
+                )
+            ))->renderWith(self::class . '/RealMeLoginButton');
+        } else {
+            // Login button
+            $loginButtonContent = ArrayData::create(array(
+                'Label' => _t('RealMeLoginForm.LOGINBUTTON', 'Login')
+            ))->renderWith(self::class . '/RealMeLoginButton');
+        }
+
+        return new FieldList(array(
```

Comment:
Use `FieldList::create`

Reply:
It'd also be good to have an extension point in here, e.g. `$this->extend('updateFormActions', $fieldList)` before returning it. May need to apply the Extensible trait to do so